HAI-3293 Fix not showing work areas in new hanke areas

umeetiusbaar opened the pull request:
# Description

Handle adding work areas to new hanke areas correctly. Do the same with hanke areas that no longer have work areas.

### Jira Issue: https://helsinkisolutionoffice.atlassian.net/browse/HAI-3293

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Other

# Instructions for testing

1. Have a hanke that has at least 3 areas
2. Have a kaivuilmoitus with a work areas in 2 hanke areas
3. Have a täydennyspyyntö and täydennys for kaivuilmoitus
4. In täydennys, remove a work area in one hanke area and add a new work area in new hanke area
5. Check that application view shows areas and haittojen hallinta correctly

# Checklist:

- [ ] I have written new tests (if applicable)
- [ ] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
      or other location:

# Other relevant info

Please describe here if there is e.g. some requirements for this change or
other info that the tester/user needs to know.

markohaarni commented:
I removed work area from one hanke area and added work area to new hanke area and this happened:
<img width="978" alt="Screenshot 2025-03-05 at 12 51 39" src="https://github.com/user-attachments/assets/e6c4d24a-a791-415e-9ed1-eb7417279c3b" />

umeetiusbaar commented:
> I removed work area from one hanke area and added work area to new hanke area and this happened:

That is strange. I now did the exact same (had 3 hanke areas 5, 6, 7 and 2 work areas in 5 anc 7). In täydennys, I removed work area from 7 and added a work area in 6. This is what I got:
![image](https://github.com/user-attachments/assets/201452f4-a27d-4855-973c-e1270de3c30c)

markohaarni commented:
> > I removed work area from one hanke area and added work area to new hanke area and this happened:
> 
> That is strange. I now did the exact same (had 3 hanke areas 5, 6, 7 and 2 work areas in 5 and 7). In täydennys, I removed work area from 7 and added a work area in 6. This is what I got:

I tried this again. Had 3 hanke areas 1, 2, 3 and one work area in 1 and one work area in 2. In täydennys if I remove work area from 1 and add work area to 3 and save and quit I get the result I had previously. Seems like `muutokset` does not include that area (Hankealue 3).

umeetiusbaar commented:
> > > I removed work area from one hanke area and added work area to new hanke area and this happened:
> > 
> > 
> > That is strange. I now did the exact same (had 3 hanke areas 5, 6, 7 and 2 work areas in 5 and 7). In täydennys, I removed work area from 7 and added a work area in 6. This is what I got:
> 
> I tried this again. Had 3 hanke areas 1, 2, 3 and one work area in 1 and one work area in 2. In täydennys if I remove work area from 1 and add work area to 3 and save and quit I get the result I had previously. Seems like `muutokset` does not include that area (Hankealue 3).

I removed the if-clause that caused the error.
